Mathlib/Data/List/Lex.lean

```diff
@@ -61,6 +61,15 @@ theorem not_nil_right (r : α → α → Prop) (l : List α) : ¬Lex r l [] :=
   fun.
 #align list.lex.not_nil_right List.Lex.not_nil_right
 
+theorem nil_left_or_nil_eq {r : α → α → Prop} (l : List α) : List.Lex r [] l ∨ [] = l :=
```

Comment:
I'd argue
```suggestion
theorem nil_left_or_eq_nil {r : α → α → Prop} (l : List α) : List.Lex r [] l ∨ l = [] :=
```
is slightly more useful, as there are more lemmas of the shape `l = []`. Feel free to leave as is if you feel strongly the other way.

bors d+